src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -12,8 +13,7 @@ internal static class MetadataSignatureHelper
         internal static BlobBuilder FieldSignatureEncoder(Type fieldType, ModuleBuilderImpl module)
         {
             BlobBuilder fieldSignature = new();
-
-            WriteSignatureForType(new BlobEncoder(fieldSignature).FieldSignature(), fieldType, module);
+            WriteSignatureForType(new BlobEncoder(fieldSignature).Field().Type(fieldType.IsByRef), fieldType, module);
```

Comment:
I think it would be better to call `Builder.WriteByte((byte)SignatureTypeCode.ByReference);` ourselves inside `WriteSignatureForType` for byrefs instead of making all callers worry about it.

Reply:
I will give it try, thanks